src/output.rs

```diff
@@ -62,9 +99,22 @@ fn print_entry_colorized<W: Write>(
     ls_colors: &LsColors,
 ) -> io::Result<()> {
     // Split the path between the parent and the last component
-    let mut offset = 0;
-    let path = entry.stripped_path(config);
-    let path_str = path.to_string_lossy();
+    let mut offset        = 0;
+    let path              = entry.stripped_path(config);
+    let mut path_str      = path.to_string_lossy();
+    let mut needs_quoting = false;
+
+    // Wrap the path in quotes
+    if config.use_quoting {
+        let tmp_str = quote_path(&path_str);
+
+        // If the quoted string is new, then we flag that to tweak the offset
+        // so the colors line up
+        if tmp_str != path_str {
```

Comment:
This has to compare every character of the string. It would be more efficient if you did something like have return `quote_path` return an `Option` that is only Some if quoting happened, and then do:

```rust
if let Some(quoted) = quote_path(&path_str) {
    path_str = quoted.into();
    needs_quoting = true;
}
```